src/Nethermind/Nethermind.Blockchain/Blocks/PotentialCensorshipCache.cs

```diff
@@ -0,0 +1,79 @@
+// SPDX-FileCopyrightText: 2023 Demerzel Solutions Limited
+// SPDX-License-Identifier: LGPL-3.0-only
+
+using Nethermind.Core;
+using Nethermind.Core.Caching;
+using Nethermind.Core.Crypto;
+using Nethermind.Int256;
+using Nethermind.TxPool.Collections;
+
+namespace Nethermind.Blockchain.Blocks;
+
+public class PotentialCensorshipCache
+{
+    private static PotentialCensorshipCache? _instance;
+
+    public static PotentialCensorshipCache Instance()
+    {
+        _instance ??= new PotentialCensorshipCache();
+        return _instance;
+    }
```

Comment:
Why singleton?

Reply:
We need singleton because we need to get the status of CensorshipDetected when sending the Payload, which is static.